src/main/java/uk/gov/ons/ctp/response/sample/service/SampleService.java

```diff
@@ -131,4 +131,20 @@ SampleSummary ingest(SampleSummary sampleSummary, MultipartFile file, String typ
    * @return
    */
   SampleUnit findSampleUnitBySampleUnitId(UUID sampleUnitId);
+
+  /**
+   * Get sample attributes by their associated post code
+   *
+   * @param postcode
+   * @return
+   */
+  List<SampleAttributes> findSampleAttributesByPostcode(String postcode);
```

Comment:
Wonder if it would be better to split this out into another class. The sampleserviceimpl is becoming a bit of a god class

Reply:
@benjefferies It's a pretty vast class. I'm not sure how we can split it out though. Given we might be integrating sample attributes into the sample unit table, the sample attributes stuff will end up finding their home here anyway.